internal/apicodec/codec.go

```diff
@@ -33,7 +33,7 @@ type Codec interface {
 	// EncodeRegionRange encode region's start and end.
 	EncodeRegionRange(start, end []byte) ([]byte, []byte)
 	// DecodeRegionRange decode region's start and end.
-	DecodeRegionRange(encodedStart, encodedEnd []byte) ([]byte, []byte, error)
+	DecodeRegionRange(encodedStart, encodedEnd []byte) (bool, []byte, []byte, error)
```

Comment:
We should return an error while the region range is invalid instead of a boolean flag.